lightning/src/ln/peer_handler.rs

```diff
@@ -1727,7 +1736,24 @@ impl<Descriptor: SocketDescriptor, CM: Deref, RM: Deref, OM: Deref, L: Deref, CM
 			if peer_lock.their_features.as_ref().unwrap().supports_gossip_queries() &&
 				!peer_lock.sent_gossip_timestamp_filter {
 				peer_lock.sent_gossip_timestamp_filter = true;
-				peer_lock.sync_status = InitSyncTracker::ChannelsSyncing(0);
+
+				#[allow(unused_mut)]
+				let mut should_do_full_sync = true;
+				#[cfg(feature = "std")]
+				{
+					// if the timestamp range starts more than six hours ago, do a full sync
+					// otherwise, only forward ad-hoc gossip
```

Comment:
This comment would be less confusing if it was re-phrased to say "less than six hours ago" as that is the condition where `should_do_full_sync` is modified below.

Reply:
It describes the opposite of the `if` clause as worded.

Reply:
Eh, I mean, I read the comment then see if the code does what it says, but I need to reverse the logic to determine that. Can't we just say: "Forward ad-hoc gossip if the timestamp range is less than six hours ago. Otherwise, do a full sync."?

Reply:
Looks the same?